### PR TITLE
added test for secondary-dropdown

### PR DESCRIPTION
**Pull request by Dishebh**
For issue: #519

### CODE REVIEW

**Comment by Dishebh** [comment]
@Marvin9 can you pls tell what other tests should I write for this?

**Comment by Marvin9** [comment]
You can use ```simulate``` event on input & then test rendered results. Refer ```defaultIconInput.test.tsx```

https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/simulate.html

**Comment by Dishebh** [comment]
> You can use `simulate` event on input & then test rendered results. Refer `defaultIconInput.test.tsx`
> 
> https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/simulate.html

I have been trying to replicate the same approach, using `simulate` to call the `onChange` on my `input` element, but for some reason, the `jest.fn()` doesn't get called. Pls have a look at my code snippet below: 

`let comp = mount(<SearchBar {...props} />);
    comp.props.handleChange = jest.fn();
    comp.update();

    wrapper.find('input').simulate('change');

    expect(comp.find('input').length).toEqual(1);
    expect(comp.props.handleChange).toHaveBeenCalled();`

I have also tried using `SearchBarWrapper` and `SecondaryDropDown`, but the error remains same,  and even the `input` value `wrapper.find('input').node.value` doesn't change after `simulate`. I have been stuck on this part since almost a day now. If you know, can you pls tell where I'm going wrong, and how should I do it the right way, @Marvin9 ?

**Comment by Marvin9** [comment]
FIrst of all ```SearchBar``` is input, So you are passing wrong prop. It should be ```onChange``` not ```handleChange```. We know input ```onChange``` always works, so no need tests for it.

Ideal test case should be
```it should render correct values when searched```
***Note: wrapper should be SecondaryDropDown***

What you could do is.

- simulate ```change``` event on ```SearchBar```.

```javascript
wrapper.find(SearchBar).simulate('change', { target: { value: 'Any false Value related to data' } } );
```

- ```expect``` what ```Avatar```  & ```Heading5``` should be? When you search for above value.

Don't worry, It is part of learning curve. Never loose motivation.

**Comment by Dishebh** [comment]
> FIrst of all `SearchBar` is input, So you are passing wrong prop. It should be `onChange` not `handleChange`. We know input `onChange` always works, so no need tests for it.
> 
> Ideal test case should be
> `it should render correct values when searched`
> _**Note: wrapper should be SecondaryDropDown**_
> 
> What you could do is.
> 
> * simulate `change` event on `SearchBar`.
> 
> ```js
> wrapper.find(SearchBar).simulate('change', { target: { value: 'Any false Value related to data' } } );
> ```
> 
> * `expect` what `Avatar`  & `Heading5` should be? When you search for above value.
> 
> Don't worry, It is part of learning curve. Never loose motivation.

It worked, now! Thanks for the suggestion! :rocket:

**Comment by Dishebh** [comment]
> Write more descriptive test case.
> 
> Add more data which are related to each other. Like `Helen`, `Helek` then simulate search for `Hele`. expect Helen and Helek as result.
> 
> It is just an example, try to cover as much as possible.

How about now?

**Comment by Dishebh** [comment]
> Good work. It would be better if you generalize one function for it.
> 
> ```js
> const searchHelper = (wrapper, searchValue, expectedResults) => {
>  // simulate searchValue
> 
> // iterate expectedResults and compare it with i-th Heading5 & Avatar
> // Avatar's source & Heading5's text
> }
> ```
> 
> ```js
> expectedResults = [
> { avatar: 'foo', heading: 'bar'  },
> ];
> ```
> 
> Now inside test case, you just required to call above function.
> 
> `searchHelper(wrapper, 'he', [{ avatar: '', heading: '' }, { avatar: '', heading: '' }])`

Done! Thanks!

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/ignitus-secondaryDropDown/Components/index.test.tsx [hunk 1]
```diff
@@ -35,6 +35,29 @@ const props = {
...
+    .simulate('change', { target: { value: `${searchValue}` } });
+  for (let i = 0; i < expectedResults.length; i += 1) {
+    expect(
+      wrapper
+        .find(Heading5)
+        .at(i)
+        .text(),
+    ).toEqual(expectedResults[i].name);
+  }
+  for (let i = 0; i < expectedResults.length; i += 1) {
```
Comment (Marvin9) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/ignitus-secondaryDropDown/Components/index.test.tsx:51]: No need to create separate loop. Place body of this loop in previous one.

**Pull request merged**